### PR TITLE
Minimal unit tests

### LINKED ISSUE #3: Unit Tests for Proof Graph Class
Write extensive unit tests that ensure the graph works properly. Make sure all methods are tested with all edge cases. 
Create a new project called /tests in root and use gdscript to test all methods.

### PR DESCRIPTION
**Pull request by PapaBrown**
For now all tests can't be ran at once. Must run each test function individually by commenting out the other functions you don't want to run in the run_tests(). Otherwise the tests work as expected.

ex: To test_addNode()
  test_addNode()
#test_addEdge()
#test_removeNode()
#test_removeEdge()
#test_removeNodeWithoutEdges()

### CODE REVIEW

File: project/Scripts/UnitTests.gd [hunk 1]
```diff
@@ -3,210 +3,524 @@ extends Node
...
+#		if i == 12:
+#			continue
+#		graph.addEdge(get_node("ProofGraph/12"), get_node("ProofGraph/i"))
+#	var graphNodes = graph.get_node_map()
+#	var twelve = graphNodes[12]
+#	for curr in graphNodes:
+#		if curr.second == twelve:
+#			assert(curr.second.neighbors.size() == 49)
+#		else:
+#			assert(curr.second.neighbors.size() == 1)
```
Comment (James-Oswald) [inline project/Scripts/UnitTests.gd]: No commented out code in master please

File: project/Scripts/UnitTests.gd [hunk 2]
```diff
@@ -3,210 +3,524 @@ extends Node
...
+#			var graphNode = graphNodes[i]
+#			assert(answer.location[0] == graphNode.location[0])
+#			assert(answer.location[1] == graphNode.location[1])
+#			assert(answer.location[2] == graphNode.location[2])
+#
+#		else:
+#			assert(graphNodes.find(i) == graphNodes.end() == true)
+#		assert(graphNodes.size() == 25)
+	
+	#test removeNodeWithoutEdges at positive locations
```
Comment (James-Oswald) [inline project/Scripts/UnitTests.gd]: CC

**Review by James-Oswald — APPROVED** [review]

**Review by James-Oswald — APPROVED** [review]

**Comment by PapaBrown** [comment]
UPDATE: All test functions can be run at the same time. Can still run tests individually by commenting out the other functions you don't want to run in run_tests(). To move around the graph, click on Camera3D in Scene, go to Inspector, click on transform and change the x, y, or z coordinate in Position.
